### PR TITLE
Big refactor testin

### PR DESCRIPTION
**Pull request by BOsterbuhr**
Added functions to `TestTransactionPeriod()` and improved `_handle_recipient_committee_type()` and `_handle_two_year_transaction_period()`

### CODE REVIEW

File: src/data/data_fetcher.py
Comment (NapsterInBlue) [inline src/data/data_fetcher.py:140]: This is elegant as hell, I like it :)

File: tests/test_url_fns.py
Comment (NapsterInBlue) [inline tests/test_url_fns.py:42]: Beautiful, man. Really great stuff!

File: src/data/data_fetcher.py
Comment (NapsterInBlue) [inline src/data/data_fetcher.py:112]: Don't need this, if we're casting as a `str` at the end

File: src/data/data_fetcher.py
Comment (BOsterbuhr) [reply src/data/data_fetcher.py:140]: Honestly when I thought of it that imposter syndrome disappeared for like an hour lol.

File: src/data/data_fetcher.py
Comment (NapsterInBlue) [inline src/data/data_fetcher.py:110]: At this point, we're all `int` until the end. I'd add a `two_year_transaction_period = int(two_year_transaction_period)` right above this to declutter calling that again and again

File: src/data/data_fetcher.py
Comment (NapsterInBlue) [inline src/data/data_fetcher.py:113]: Taking a step back, the fact that we have `print("Invalid input, defaulting to 2020.")` in multiple locations should set your spidey senses a'tingling that there's a cleaner way to write this.

Few leading questions for you:

1. Does this line need to be intented?
2. Do we *need* `else` statements?
3. Can we simplify our flow control by:
    * Inverting our check (year **in** the range vs year **out of** range)
    * Can we make good use of a mid-function `return` statement?

Consider the following pseudocode

```
if is_numeric
    if is_odd_year
       make_it_appropriate_even_year
    if in_bounds
       return str(value)
error_msg
return '2020'
```

Any records you pass in within the range won't ever make it to the `error_msg` line.

But don't take my word for it-- we wrote unit tests precisely so we can tinker on this function and be confident we're on the right track :)

**Review by NapsterInBlue — CHANGES_REQUESTED** [review]

File: src/data/data_fetcher.py
Comment (NapsterInBlue) [reply src/data/data_fetcher.py:140]: Be proud of it, man. I look back at my old code and cringe like hell (years old, months old, weeks old. Doesn't matter.)

Don't see any room for improvement on this one 🔥

**Comment by BOsterbuhr** [comment]
Ok so there are a few thing wrong with this still but not in reference to the things we have addressed. I have everything fixed but also some new features, so I am going to wait until we merge this pull request to make another branch. Is that the proper workflow?

**Comment by NapsterInBlue** [comment]
Sure is! And this all looks great, man :)

Obviously, I'm... marginally less-available to hack on stuff during the week, but open up issues if you want to hash out planning/design. I'm happy to weigh in as best I can!

**Pull request merged**
